### PR TITLE
Code fixes for MSVC

### PR DESCRIPTION
**Pull request by ns6089**
## Description

A number of standard-compliance and portability fixes required for compilation in MSVC.

Doesn't enable compilation in MSVC out of the box, Changes to submodules (nanors in particular) and CMakeLists.txt are needed, as well as adding vcpkg. Will be done in later PRs.

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

**Comment by ns6089** [comment]
Don't review yet.
Will try replacing `std::to_address()` with `std::addressof()` and bring the compiler requirement back to c++17.
Hopefully this will make every platform toolchain happy.

File: src/platform/windows/PolicyConfig.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 #include <mmdeviceapi.h>
 
-#ifdef __MINGW32__
+#ifdef _WIN32
```
Comment (cgutman) [inline src/platform/windows/PolicyConfig.h:13]: This is in `platform/windows` so `_WIN32` should always be defined when building it.

File: src/platform/windows/publish.cpp [hunk 2]
```diff
@@ -29,7 +31,7 @@ using namespace std::literals;
 #define SV(quote) __SV(quote)
 
 extern "C" {
-#ifndef __MINGW32__
+#ifndef _WIN32
```
Comment (cgutman) [inline src/platform/windows/publish.cpp:34]: Similar to case in PolicyConfig. If we don't need them with `_WIN32` (either MinGW or MSVC), you can just remove them.

**Comment by ns6089** [comment]
> Will try replacing std::to_address() with std::addressof() and bring the compiler requirement back to c++17.
Hopefully this will make every platform toolchain happy.

Doesn't work that well, dereferencing end iterator triggers msvc runtime checks in debug mode. Back to draft.

**Pull request closed without merging**